bindings/pydeck/pydeck/bindings/types.py

```diff
@@ -0,0 +1,101 @@
+import abc
+import json
+from collections.abc import Iterable
+
+TYPE_IDENTIFIER = "@@type"
+FUNCTION_IDENTIFIER = "@@="
+ENUM_IDENTIFIER = "@@#"
+
+
+class Type(metaclass=abc.ABCMeta):
```

Comment:
I would suggest naming this something more descriptive than "Type", which seems a little generic.

Reply:
Since it's a base class, `BaseType` might be more apt?

Reply:
Nit: Naming base classes with `Base...` is one my pet peeves. I feel that it is a design cop-out. Let's ideate a bit more first.

`SerializableType`, `TransportType`, `TransportableType`, ...

Reply:
Right. This base class isn't public so I'm not too concerned with what the name should be, and welcome those suggestions. I'd also consider prefixing with `_` to make clear it's private.